### PR TITLE
[develop-poc] Fixes + CI/CD

### PR DESCRIPTION
**Pull request by leonkozlowski**
**NEW**
-Added `tox.ini`
-Added `.travis.yml`
-Added `Makefile` for build in future
-Added `CHANGELOG.md` -
-Added `setup.py` -
-Added `MANIFEST.in` for travis integration

**ENHANCEMENTS**
-Introduced baseline state tracking with `src.jiggy.state.State`
-Added a wrapper for function runs in `SequentialRunner` to increase abstraction

```python
    @staticmethod
    def __cls_run(init_cls: getattr, argument=None):
        """Initialize function wrapper with state tracking."""
        state = State.PENDING
        try:
            if argument:
                output = init_cls.run(argument)
            else:
                output = init_cls.run()
            state = State.SUCCESS
        except Exception:
            output = None
            state = State.FAILED

        return state, output
```

This has very base level state tracking and exception handling - pipeline can continue without deps

**FIXES**
-I re-added `Task` as there were dependencies on it that I will explain:
Basically any Task that gets created will be instantiated as a Task with a `run()` - which then is instantiated as a `JagTask` which I also re-added, which in turn gives up some good abstraction for what is code and what is a task in our pipeline.
-Also fixed some filepaths in some of the `*.yml` for integration test

**MINIMAL TEST CASE**
Minimal test case.

```python
if __name__ == '__main__':
    myrun = SequentialRunner("examples/inputs/no_ext.yml").run()
    
    >>> myrun
    >>> ['<<Task `get-current-date`>, SUCCESS>', '<<Task `get-weekday`>, SUCCESS>', '<<Task `get-first-letter`>, SUCCESS>']
```

*EDIT*
c683bcd: mv `JagTask` -> `JigTask`

### CODE REVIEW

**Comment by mitchbregs** [comment]
Quick question before we merge this - is the assumption that when someone implements a `jiggy.Task` they will need to specify `self._name`? Out of user friendliness, `self.name` or a class variable like Luigi would be a bit nicer. Let's merge this in and discuss later. LGMT +1 👍

**Review by mitchbregs — APPROVED** [review]

**Comment by leonkozlowski** [comment]
> Quick question before we merge this - is the assumption that when someone implements a `jiggy.Task` they will need to specify `self._name`? Out of user friendliness, `self.name` or a class variable like Luigi would be a bit nicer. Let's merge this in and discuss later. LGMT +1 👍

Name eventually gets set in `JigTask`, since its inherited, it can be taken out of `jiggy.Task`

**Comment by mitchbregs** [comment]
> > Quick question before we merge this - is the assumption that when someone implements a `jiggy.Task` they will need to specify `self._name`? Out of user friendliness, `self.name` or a class variable like Luigi would be a bit nicer. Let's merge this in and discuss later. LGMT +1 👍
> 
> Name eventually gets set in `JigTask`, since its inherited, it can be taken out of `jiggy.Task`

Let's merge this, awesome stuff.
